test/integration/scheduler/preemption/preemption_test.go

```diff
@@ -454,6 +458,145 @@ func TestPreemption(t *testing.T) {
 	}
 }
 
+// TestPreemptionWithJobBackoffPolicyGate tests preemption with the JobBackoffPolicy feature enabled
+func TestPreemptionWithJobBackoffPolicyGate(t *testing.T) {
```

Comment:
Instead of creating a new test, could you add a test case in TestPreemption where the feature is enabled?

Reply:
Yes, although if we go this way, I would like to simplify the existing test:
- eliminate the scheduler from the pre-existing test
- use wait.PollImmidiately instead of custom "waitForPodCondition" loop
- use "node.kubernetes.io/exclude-disruption" instead of periodic heart beats because it greatly simplifies the code test. This is in fact suggested in one comment in the test:
```golang
	// Regularly send heartbeat event to APIServer so that the cluster doesn't enter fullyDisruption mode.
	// TODO(Huang-Wei): use "NodeDisruptionExclusion" feature to simply the below logic when it's beta.
```

Reply:
> eliminate the scheduler from the pre-existing test

this is a scheduler test, we need the scheduler :)

The rest sounds good to me. Could you do it in another PR?

Reply:
Oh, when I was writing my answer I was thinking about the pre-existing test in lifecycle_test. I created an issue to simplify it (https://github.com/kubernetes/kubernetes/issues/111183). 

As for the scheduler test the mentioned issues are irrelevant so it should be easy to integrate it with the existing tests. Going to try.

Reply:
merged